WireDomain/Sources/WireDomain/Components/ClientSessionComponent.swift

```diff
@@ -0,0 +1,305 @@
+
+    private lazy var teamsAPI: any TeamsAPI = TeamsAPIBuilder(apiService: apiService).makeAPI(for: apiVersion)
+
+    private lazy var updateEventsAPI: any UpdateEventsAPI = UpdateEventsAPIBuilder(apiService: apiService)
+        .makeAPI(for: apiVersion)
+
+    private lazy var userClientsAPI: any UserClientsAPI = UserClientsAPIBuilder(apiService: apiService)
+        .makeAPI(for: apiVersion)
+
+    private lazy var userConnectionsAPI: any ConnectionsAPI = ConnectionsAPIBuilder(apiService: apiService)
+        .makeAPI(for: apiVersion)
+
+    private lazy var usersAPI: any UsersAPI = UsersAPIBuilder(apiService: apiService).makeAPI(for: apiVersion)
+
+    private lazy var userPropertiesAPI: any UserPropertiesAPI = UserPropertiesAPIBuilder(apiService: apiService)
+        .makeAPI(for: apiVersion)
+
+    // MARK: - Local storage
+
+    private lazy var backendConfigLocalStore: some BackendConfigLocalStoreProtocol = BackendConfigLocalStore(
+        sharedUserDefaults: sharedUserDefaults
+    )
+
+    private lazy var conversationLabelsLocalStore: some ConversationLabelsLocalStore = ConversationLabelsLocalStore(
+        context: syncContext
+    )
+
+    private lazy var conversationLocalStore: some ConversationLocalStoreProtocol = ConversationLocalStore(
+        context: syncContext,
+        mlsService: mlsService,
+        userLocalStore: userLocalStore,
+        messageLocalStore: messageLocalStore
+    )
+
+    private lazy var featureConfigsLocalStore: some FeatureConfigLocalStoreProtocol = FeatureConfigLocalStore(
+        context: syncContext
+    )
+
+    private lazy var messageLocalStore: some MessageLocalStoreProtocol = MessageLocalStore(
+        context: syncContext,
+        userLocalStore: userLocalStore
+    )
+
+    private lazy var teamLocalStore: some TeamLocalStoreProtocol = TeamLocalStore(
+        context: syncContext,
+        userLocalStore: userLocalStore
+    )
+
+    private lazy var updateEventsLocalStore: some UpdateEventsLocalStoreProtocol = UpdateEventsLocalStore(
+        context: eventContext,
+        userID: selfUserID,
+        sharedUserDefaults: sharedUserDefaults
+    )
+
+    private lazy var userClientsLocalStore: some UserClientsLocalStore = UserClientsLocalStore(
+        context: syncContext,
+        userLocalStore: userLocalStore
+    )
+
+    private lazy var userConnectionsStore: some ConnectionsLocalStoreProtocol = ConnectionsLocalStore(
+        context: syncContext
+    )
+
+    private lazy var userLocalStore: some UserLocalStoreProtocol = UserLocalStore(
+        context: syncContext,
+        userDefaults: sharedUserDefaults
+    )
+
+    // MARK: - Pull syncs
+
+    private lazy var pullAllConversationsSync: some PullAllConversationsSyncProtocol = PullAllConversationsSync(
+        localDomain: localDomain,
+        isFederationEnabled: BackendInfo.isFederationEnabled,
+        isMLSEnabled: BackendInfo.isMLSEnabled,
```

Comment:
question: so if the backendInfo change, I guess we tore down the clientSessionComponent

Reply:
Yeah, at the moment it's static, so probably we should rebuild the session. But I also think that we should only resolve backed information once during the lifetime of the app to simplify things. What do you think?